### PR TITLE
Additional Commentary for JS templates

### PR DESCRIPTION
**Pull request by kingthorin**
Add JSDoc style comments to base functions in:

Active default template.js
Passive default template.js
Proxy default template.js
Targeted default template.js

Other script types seemed to have sufficiently complete commentary.

Fixes zaproxy/zaproxy#1107

### CODE REVIEW

File: src/scripts/templates/active/Active default template.js [hunk 1]
```diff
@@ -4,7 +4,14 @@
 // Note that new active scripts will initially be disabled
 // Right click the script in the Scripts tree and select "enable"  
 
-
+/**
+ * Scans a "node", i.e. an individual entry in the Site Tree.
```
Comment (thc202) [inline src/scripts/templates/active/Active default template.js]: "Sites" instead?

Guess the first two lines of the file could be deleted, replaced by new docs?

File: src/scripts/templates/active/Active default template.js [hunk 2]
```diff
@@ -4,7 +4,14 @@
...
+ * 
+ * @param as the ActiveScan parent object that will do all the core interface tasks
```
Comment (thc202) [inline src/scripts/templates/active/Active default template.js]: The actual class is ScriptsActiveScanner (which is itself a scanner).

File: src/scripts/templates/active/Active default template.js [hunk 3]
```diff
@@ -36,6 +43,16 @@ function scanNode(as, msg) {
 	}
 }
 
+/**
+ * Scans a specific parameter in an HTTP message.
+ * 
+ * @param as the ActiveScan parent object that will do all the core interface tasks
```
Comment (thc202) [inline src/scripts/templates/active/Active default template.js]: Same here.

File: src/scripts/templates/active/Active default template.js [hunk 4]
```diff
@@ -36,6 +43,16 @@ function scanNode(as, msg) {
...
 }
 
+/**
+ * Scans a specific parameter in an HTTP message.
+ * 
+ * @param as the ActiveScan parent object that will do all the core interface tasks 
+ *     (i.e.: sending and receiving messages, providing access to Strength and Threshold settings,
+ *     raising alerts, etc.)
+ * @param msg the HTTP Message being scanned
+ * @param param the parameter being manipulated for this test/scan
```
Comment (thc202) [inline src/scripts/templates/active/Active default template.js]: Could mention that it's the name of the parameter. Also, could indicate the type (string) per JSDoc? [1]

[1] http://usejsdoc.org/tags-param.html#names-types-and-descriptions

File: src/scripts/templates/active/Active default template.js [hunk 5]
```diff
@@ -36,6 +43,16 @@ function scanNode(as, msg) {
...
+ * @param value the original parameter value retrieved by the proxy or spider
```
Comment (thc202) [inline src/scripts/templates/active/Active default template.js]: Is the "retrieved" part really needed? That might not be the case (messages manually sent, other tools).

File: src/scripts/templates/passive/Passive default template.js [hunk 6]
```diff
@@ -4,6 +4,14 @@
 // Note that new passive scripts will initially be disabled
 // Right click the script in the Scripts tree and select "enable"  
 
+/**
+ * Passively scans an HTTP message.
+ * 
+ * @param ps the PassiveScan parent object that will do all the core interface tasks
```
Comment (thc202) [inline src/scripts/templates/passive/Passive default template.js]: In this case the object is ScriptsPassiveScanner, also a scanner.

File: src/scripts/templates/active/Active default template.js (hunk 2 above)
Comment (kingthorin) [inline src/scripts/templates/active/Active default template.js]: Hmm ok I'll try to revise to be more specific, trying to hit a balance between dev comments and user comments :smiley:

File: src/scripts/templates/active/Active default template.js (hunk 4 above)
Comment (kingthorin) [inline src/scripts/templates/active/Active default template.js]: Good call

**Comment by kingthorin** [comment]
All addressed

File: src/scripts/templates/passive/Passive default template.js [hunk 7]
```diff
@@ -1,9 +1,18 @@
-// The scan function will be called for request/response made via ZAP, excluding some of the automated tools
 // Passive scan rules should not make any requests 
 
 // Note that new passive scripts will initially be disabled
 // Right click the script in the Scripts tree and select "enable"  
 
+/**
+ * Passively scans an HTTP message. // The scan function will be called for
```
Comment (thc202) [inline src/scripts/templates/passive/Passive default template.js]: The // could be removed.

**Comment by kingthorin** [comment]
Look good now?

**Comment by thc202** [comment]
Yeah, looks good to me, thank you! (I didn't notice your update, no email notification.)

**Comment by kingthorin** [comment]
No problem, I'm not worried. Things can easily get buried/missed when lots of things are moving/updating/changing... Thanks

**Pull request merged**